### PR TITLE
feat(STONEINTG-523): enum for integration test status

### PR DESCRIPTION
**Pull request by MartinBasti**
This commit provides base data structure for the keeping status of tests.

Type supports string and JSON interfaces to convert between numeric and textual representation.

## Maintainers will complete the following section

- [x] Commit messages are descriptive enough ([hints](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/))
- [x] Code coverage from testing does not decrease and new code is covered
- [x] [Controllers diagrams](https://github.com/redhat-appstudio/integration-service/tree/main/docs) are updated when PR changes controllers code  (if applicable)

### CODE REVIEW

File: gitops/snapshot.go [hunk 1]
```diff
@@ -117,6 +117,25 @@ const (
 	AppStudioIntegrationStatusFinished = "Finished"
 )
 
+// IntegrationTestScenario test runs status
+type TestStatus int
```
Comment (MartinBasti) [inline gitops/snapshot.go:121]: Feel free to suggest better name

File: gitops/snapshot.go [hunk 2]
```diff
@@ -117,6 +117,25 @@ const (
...
+
+//go:generate enumer -type=TestStatus -linecomment -json
```
Comment (hongweiliu17) [inline gitops/snapshot.go:123]: The usage looks great. I didn't know it before, let me search a doc to learn. Thanks

File: gitops/snapshot.go (hunk 1 above)
Comment (MartinBasti) [reply gitops/snapshot.go]: updated

**Comment by MartinBasti** [comment]
Coverage is low because of generated code, can I wave it?

**Comment by dirgim** [comment]
> Coverage is low because of generated code, can I wave it?

I think it would be best to add the generated file to the ignore list in [codecov.yml](https://github.com/redhat-appstudio/integration-service/blob/main/codecov.yml).

**Review by hongweiliu17 — APPROVED** [review]
LGTM

**Review by dirgim — APPROVED** [review]
LGTM

**Pull request merged**